atoma-sui/src/subscriber.rs

```diff
@@ -261,6 +261,84 @@ impl SuiEventSubscriber {
     }
 }
 
+/// Reads an event cursor from a TOML file.
+///
+/// This function attempts to read and parse an event cursor from the specified file path.
+/// If the file doesn't exist, it will create an empty file and return `None`. If the file
+/// exists, it will attempt to parse its contents as an `EventID`.
+///
+/// # Arguments
+///
+/// * `path` - A string slice containing the path to the TOML file
+///
+/// # Returns
+///
+/// * `Result<Option<EventID>>` - Returns:
+///   * `Ok(Some(EventID))` if the file exists and was successfully parsed
+///   * `Ok(None)` if the file doesn't exist (and was created)
+///   * `Err(SuiEventSubscriberError)` if:
+///     * The file exists but couldn't be read
+///     * The file contents couldn't be parsed as TOML
+///     * The file couldn't be created when not found
+///
+/// # Examples
+///
+/// ```no_run
+/// let path = "cursor.toml";
+/// match read_cursor_from_toml_file(path) {
+///     Ok(Some(cursor)) => println!("Read cursor: {:?}", cursor),
+///     Ok(None) => println!("No cursor found, created empty file"),
+///     Err(e) => eprintln!("Error reading cursor: {}", e),
+/// }
+/// ```
+fn read_cursor_from_toml_file(path: &str) -> Result<Option<EventID>> {
+    match std::fs::read_to_string(path) {
+        Ok(content) => {
+            let cursor: EventID = toml::from_str(&content)?;
+            Ok(Some(cursor))
+        }
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => {
+            // Create the file if it doesn't exist
+            std::fs::write(path, "")?;
```

Comment:
if you don't plan to write anything into the file, then use `std::fs::File::create_new(path)`

Reply:
I removed this, as reading from empty file doesn't require creating the file, instead we should write the file for the first on shutdown (I reckon). What are your thoughts ?

Reply:
on shutdown? shouldn't we create the file only when needed?